R/download_file_linter.R

```diff
@@ -40,14 +40,36 @@ download_file_linter <- function() {
 
     implicit_mode <- is.na(download_file_mode)
 
+    # According to download.file() documentation, mode is ignored when method
+    # is "curl" or "wget" so if it's explicitly specified, we should let the
+    # user know they might not get the desired effect
+    download_file_method <- get_r_string(
+      download_file_calls,
+      "parent::expr[1]/
+        SYMBOL_SUB[text() = 'method']/
+        following-sibling::expr[1]/STR_CONST
+      "
+    )
+    no_mode_method <- !is.na(download_file_method) & download_file_method %in% c("wget", "curl")
```

Comment:
naming suggestion: `has_ignored_mode`. Particularly would like to avoid `!no` double-negation